books/kestrel/c/syntax/validator.lisp

```diff
@@ -1276,29 +1276,23 @@
      an lvalue as left operand [C17:6.5.16/2],
      but currently we do not check that.
      In our currently approximate type system,
-     the requirements in [C17:6.5.16.1/1] reduce to the two operands having
-     both arithmetic types,
-     or both the structure type,
-     or both the union type,
-     or both pointer types,
-     or one pointer type and one integer type.
-     We do not perform array-to-pointer or function-to-pointer conversion
-     on the left operand, because the result would not be an lvalue.
-     In order to allow a pointer type on the left
-     and a null pointer constant [C17:6.3.2.3/3] without @('void *') cast
-     on the right,
-     since we do not have code to recognize such null pointer constants yet,
-     we allow any integer expressions on the right
-     when the expression on the left has pointer type.
-     We also allow the left operand to be boolean
-     and the right operand to be a pointer;
-     although [C17:6.5.9/2] does not mention that,
-     we found it accepted by practical compiler with a strict C17 option,
-     and [C17:6.3.1.2/1] mentions a conversion
-     from scalars (which includes pointer) to booleans,
-     although it does not say that it happens
-     (it says ``When any scalar valu is converted to @('_Bool'), ...'').
-     The type of the result is the type of the left operand [C17:6.5.16/3].")
+     the requirements in [C17:6.5.16.1/1] reduce to the following four cases.")
```

Comment:
I think it would be good to keep the sentence two lines above about the type of the result. I think it probably got deleted unintentionally.

Reply:
You mean the sentence "The type of the result is the type of the left operand [C17:6.5.16/3]."? Your are right, I didn't intend to remove that; I'll add it back in.

Reply:
Yes, I meant that one. Thanks!